### PR TITLE
Adding in logging to the Zuora library 

### PR DESCRIPTION
**Pull request by charleycampbell**
## What does this change?
Adding in logging to the Zuora library to log the response and request bodies when hitting the subscriptionGetResponse and subscriptionUpdateResponse. These logs will help with looking into the issue that is occurring around holiday stops and negative invoices incorrectly being created.

## How can we measure success?
We will need to check in Cloudwatch to make sure that it is logging as expected when in production

### CODE REVIEW

File: lib/zuora-core/src/main/scala/com/gu/zuora/Zuora.scala [hunk 1]
```diff
@@ -51,17 +51,21 @@ object Zuora {
...
   def subscriptionGetResponse(config: ZuoraConfig, accessToken: AccessToken, backend: SttpBackend[Identity, Any])(subscriptionName: SubscriptionName): ZuoraApiResponse[Subscription] = {
-    basicRequest.get(uri"${config.baseUrl}/subscriptions/${subscriptionName.value}")
+    val response = basicRequest.get(uri"${config.baseUrl}/subscriptions/${subscriptionName.value}")
       .header("Authorization", s"Bearer ${accessToken.access_token}")
       .response(asJson[Subscription])
       .mapResponse(_.left.map(e => ZuoraApiFailure(e.getMessage)))
       .send(backend)
-      .body
+
+    logger.info(s"subscriptionGetResponse for sub ${subscriptionName.value} is ${response.body.toString}")
```
Comment (rupertbates) [inline lib/zuora-core/src/main/scala/com/gu/zuora/Zuora.scala:60]: You might get more readable output with something like 
```suggestion
    import io.circe.syntax._ //stick this at the top of the file
    logger.info(s"subscriptionGetResponse for sub ${subscriptionName.value} is ${response.body.asJson}")
```

File: lib/zuora-core/src/main/scala/com/gu/zuora/Zuora.scala [hunk 2]
```diff
@@ -74,7 +78,9 @@ object Zuora {
           else Left(ZuoraApiFailure(errMsg(status.reasons.map(_.mkString).getOrElse(""))))
       }
       .send(backend)
-      .body.left.map(failure => ZuoraApiFailure(errMsg(failure.reason)))
+      logger.info(s"response for subscriptionUpdateResponse is ${response.body.toString}")
```
Comment (rupertbates) [inline lib/zuora-core/src/main/scala/com/gu/zuora/Zuora.scala:81]: Same as above

**Review by rupertbates — APPROVED** [review]
Have made one suggestion but otherwise 👍

**Comment by charleycampbell** [comment]
Closing pull request as the test on DEV to check the logging was unsuccessful and the ticket has been moved back onto the backlog

**Pull request closed without merging**